tests/foreman/api/test_hostcollection.py

```diff
@@ -15,446 +15,450 @@
-        """Create host collection with different values of 'unlimited hosts'
-        parameter.
-
-        :id: d385574e-5794-4442-b6cd-e5ded001d877
-
-        :expectedresults: The host collection was successfully created and has
-            appropriate 'unlimited hosts' parameter value.
-
-        :CaseImportance: Critical
-        """
-        for unlimited in (True, False):
-            with self.subTest(unlimited):
-                host_collection = entities.HostCollection(
-                    max_hosts=1 if not unlimited else None,
-                    organization=self.org,
-                    unlimited_hosts=unlimited,
-                ).create()
-                self.assertEqual(host_collection.unlimited_hosts, unlimited)
-
-    @pytest.mark.tier1
-    def test_positive_create_with_host(self):
-        """Create a host collection that contains a host.
-
-        :id: 9dc0ad72-58c2-4079-b1ca-2c4373472f0f
-
-        :expectedresults: The host collection can be read back, and it includes
-            one host.
-
-        :CaseImportance: Critical
+@pytest.mark.tier1
+def test_positive_create_with_limit(module_org):
+    """Create host collections with different limits.
 
-        :BZ: 1325989
-        """
+    :id: 86d9387b-7036-4794-96fd-5a3472dd9160
+
+    :expectedresults: The host collection was successfully created and has
+        appropriate limit.
+
+    :CaseImportance: Critical
+    """
+    for _ in range(5):
+        limit = randint(1, 30)
         host_collection = entities.HostCollection(
-            host=[self.hosts[0]], organization=self.org
+            max_hosts=limit, organization=module_org
         ).create()
-        self.assertEqual(len(host_collection.host), 1)
+        assert host_collection.max_hosts == limit
 
-    @pytest.mark.tier1
-    def test_positive_create_with_hosts(self):
-        """Create a host collection that contains hosts.
 
-        :id: bb8d2b42-9a8b-4c4f-ba0c-c56ae5a7eb1d
+@pytest.mark.parametrize("unlimited", [False, True])
+@pytest.mark.tier1
+def test_positive_create_with_unlimited_hosts(module_org, unlimited):
+    """Create host collection with different values of 'unlimited hosts'
+    parameter.
 
-        :expectedresults: The host collection can be read back, and it
-            references two hosts.
+    :id: d385574e-5794-4442-b6cd-e5ded001d877
 
-        :CaseImportance: Critical
+    :parametrized: yes
 
-        :BZ: 1325989
-        """
-        host_collection = entities.HostCollection(host=self.hosts, organization=self.org).create()
-        self.assertEqual(len(host_collection.host), len(self.hosts))
+    :expectedresults: The host collection was successfully created and has
+        appropriate 'unlimited hosts' parameter value.
 
-    @pytest.mark.tier2
-    def test_positive_add_host(self):
-        """Add a host to host collection.
+    :CaseImportance: Critical
+    """
+    host_collection = entities.HostCollection(
+        max_hosts=1 if not unlimited else None,
```

Comment:
Optional, but consider making this statement a bit more readable by making the ternary statement 'positive':

`max_hosts = None if unlimited else 1`

Reply:
OK